sonar-db/src/main/java/org/sonar/db/charset/MssqlCharsetHandler.java

```diff
@@ -46,97 +42,81 @@
   private static final String BIN = "BIN";
   private static final String BIN2 = "BIN2";
 
-  protected MssqlCharsetHandler(SqlExecutor selectExecutor) {
+  private final Metadata metadata;
+
+  MssqlCharsetHandler(SqlExecutor selectExecutor) {
     super(selectExecutor);
+    this.metadata = new Metadata();
```

Comment:
`Metadata` could then be renamed to `MsSqlMetadataReader`, since it always "reads" metadata from a connection